torch/onnx/verification.py

```diff
@@ -731,3 +746,567 @@ def verify(
+    )
+    ort_outs = _run_ort(ort_session, ort_inputs)
+
+    has_mismatch = False
+    try:
+        _compare_ort_pytorch_outputs(
+            ort_outs=ort_outs,
+            pt_outs=jit_outs,
+            rtol=rtol,
+            atol=atol,
+            check_shape=check_shape,
+            check_dtype=check_dtype,
+            ignore_none=ignore_none,
+            acceptable_error_percentage=acceptable_error_percentage,
+        )
+    except AssertionError as e:
+        print("Has mismatch: ", e)
+        has_mismatch = True
+
+    return has_mismatch, jit_outs
+
+
+@dataclasses.dataclass
+class GraphInfo:
+    graph: torch.Graph
+    input_args: Tuple[Any, ...]
+    params_dict: Dict[str, Any]
+    export_options: _experimental.ExportOptions = dataclasses.field(
+        default_factory=_experimental.ExportOptions
+    )
+    has_mismatch: Optional[bool] = None
+    pt_outs: Optional[Union[_NumericType, Sequence[_NumericType]]] = None
+    upper_graph_info: Optional[GraphInfo] = None
+    lower_graph_info: Optional[GraphInfo] = None
+    tree_tag: str = dataclasses.field(default="")
+    _excluded_node_kinds: FrozenSet[str] = dataclasses.field(
+        default_factory=lambda: frozenset({"prim::Constant", "prim::ListConstruct"}),
+    )
+
+    def clear(self):
+        self.has_mismatch = None
+        self.pt_outs = None
+        self.upper_graph_info = None
+        self.lower_graph_info = None
+
+    def essential_node_count(self) -> int:
+        return sum(
+            1 for n in self.graph.nodes() if n.kind() not in self._excluded_node_kinds
+        )
+
+    def mismatch_partition(self) -> Optional[GraphInfo]:
+        if not self.has_mismatch:
+            return None
+        if self.upper_graph_info is not None and self.upper_graph_info.has_mismatch:
+            return self.upper_graph_info.mismatch_partition()
+        if self.lower_graph_info is not None and self.lower_graph_info.has_mismatch:
+            return self.lower_graph_info.mismatch_partition()
+        if self.upper_graph_info is None and self.lower_graph_info is None:
+            return self
+        raise RuntimeError("Mismatch detected, but cannot locate the mismatching op.")
+
+    def _graph_partition_pivot(self) -> int:
+        """Find the pivot to partition the graph.
+
+        The pivot is the node that splits the graph into two parts. Each part should
+        have the same amount of nodes, excluding primitive ops, such as `prim::Constant`.
+        If the graph has an odd number of nodes, the upper part will have one more node.
+        If the graph does not have any node that can be partitioned, return -1.
+
+        Returns:
+            The index of the pivot node.
+        """
+        non_const_indices = [
+            i
+            for i, n in enumerate(self.graph.nodes())
+            if n.kind() not in self._excluded_node_kinds
+        ]
+        half_idx = len(non_const_indices) // 2 - 1
```

Comment:
are you using median? if so, do you need to sort the list of ints to get the middle value?